core-bundle/src/Routing/RouteProvider.php

```diff
@@ -206,7 +203,7 @@ private function addRoutesForRootPage(PageModel $page, array &$routes): void
             $route->getMethods()
         );
 
-        if (!$urlPrefix || (!$this->legacyRouting && $page->disableLanguageRedirect)) {
+        if (!$urlPrefix || (!$this->legacyRouting && $page->loadDetails() && $page->disableLanguageRedirect)) {
```

Comment:
`PageModel::loadDetails()` does not return a boolean, not sure if we should put that in there?

Reply:
it returns the page itself, which "equals" to true. It's only meant to inline-load the details before `disableLanguageRedirect` is queried

Reply:
```suggestion
        if (!$urlPrefix || (!$this->legacyRouting && $page->loadDetails()->disableLanguageRedirect)) {
```
Maybe something like this makes that more clear?

Reply:
good idea, changed in 018aa0e3ee56f2b18139f6eaa876805b744f679c. That also solves the phpstan issue 😆